charts/hono/values.yaml

```diff
@@ -823,107 +823,247 @@ deviceRegistryExample:
+    # should be deployed and used.
+    enabled: false
+    # imageName contains the name (including registry and tag)
+    # of the container image to use for the example Device Registry
+    imageName: index.docker.io/eclipse/hono-service-device-registry-mongodb
+    # imageTag contains the tag of the container image to deploy.
+    # If not specified, the value of the honoImagesTag property is used.
+    # imageTag:
+    # javaOptions contains options to pass to the JVM when starting
+    # up the service
+    javaOptions: -XX:MinRAMPercentage=80 -XX:MaxRAMPercentage=80
+    # resources contains the container's 'requests and limits for CPU and memory
+    # as defined by the Kubernetes API.
+    # Refer to https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/
+    # for a description of the properties' semantics.
+    resources:
+      requests:
+        cpu: "100m"
+        memory: "300Mi"
+      limits:
+        cpu: "1"
+        memory: "300Mi"
 
-    # healthCheck contains configuration properties for the service's
-    # health check server as defined by
-    # https://www.eclipse.org/hono/docs/admin-guide/monitoring-tracing-config/#health-check-server-configuration
-    # If not set, a TLS secured health check server is configured listening on
-    # all network interfaces on port 8088 using the service's
-    # example key and cert.'
-    healthCheck:
-    #  bindAddress: 0.0.0.0
-    #  port: 12000
-    #  keyPath: "/etc/config/key.pem"
-    #  certPath: "/etc/config/cert.pem"
+    svc:
+      annotations: {}
+      loadBalancerIP:
+
+    # hono contains the Device Registry's configuration properties as defined in
+    # https://www.eclipse.org/hono/docs/admin-guide/mongodb-device-registry-config/
+    hono:
+      # auth contains Hono client properties used by the example registry for
+      # connecting to the Authentication service.
+      # If not set, the registry by default uses the Auth Server component to
+      # authenticate clients.
+      auth:
+      registry:
+        # amqp contains configuration properties for the server's
+        # exposed AMQP endpoints.
+        # If not set, the registry by default exposes the secure port
+        # using an example key and certificate.
+        amqp:
+        #  insecurePortEnabled: true
+        #  insecurePortBindAddress: "0.0.0.0"
+
+        # rest contains configuration properties for the server's
+        # exposed HTTP endpoints.
+        # If not set, the registry by default exposes the insecure and secure ports
+        # using an example key and certificate.
+        http:
+        #  insecurePortEnabled: true
+        #  insecurePortBindAddress: "0.0.0.0"
+      # mongodb contains the configuration properties to connect to the MongoDB database instance.
+      # If you would like to use an already existing MongoDB database instance, then configure
+      # the below section accordingly.
```

Comment:
it looks like all of this is the same as for the file based registry, I'd rather have a single set of properties to configure the endpoints in order to make it less error prone and keep the properties and comments consistent ...

Reply:
Not all are same as of the file based registry. For example it is `http` in MongoDB registry and in case of file-based it is `rest`. And MongoDB registry  has some extra properties (with prefix `mongodb`)  to configure the database connection. In future we will have more (eg. JDBC based) device registry implementations and putting all properties under one section IMHO would be difficult and not easy to configure. Also I thought it may be easier for users who wants to use either MongoDB or file-based registry to focus on individual section, makes changes there and need not worry about the other registries. WDYT?

Reply:
They will all expose endpoints via HTTP and AMQP, won't they? And they will all need to authenticate clients on the AMQP endpoint using the Authentication service. So there is little need to repeat these config properties for all of the registries, is there?

Reply:
I see your point to reduce the repetition and it has its advantages too. Just to ensure that my understanding is correct, the idea is to have one common section for configuring the properties that are common. The rest of the properties such as `data`, image details will be in separate sections (MongoDb and file) as below.

```
deviceRegistryExample:
  enabled: true
  type: mongodb / file
  javaOptions: -XX:MinRAMPercentage=80 -XX:MaxRAMPercentage=80
  hono:
    auth:
    registry:
      amqp:
      rest:
  resources:
    requests:
      cpu: "100m"
      memory: "300Mi"
    limits:
      cpu: "1"
      memory: "300Mi"
  svc:
    annotations: {}
    loadBalancerIP:
  healthCheck:

  fileBasedDeviceRegistry:
    imageName: index.docker.io/eclipse/hono-service-device-registry-file
    data:
      volumeName:
      tenantsFile: example-tenants.json
      devicesFile: example-device-identities.json
      credentialsFile: example-credentials.json
      resetOnStartup: true
    storageClass:

  mongoDBBasedDeviceRegistry:
    imageName: index.docker.io/eclipse/hono-service-device-registry-mongodb
    hono:
      mongodb:
```